docs/source/manual/validation.rst

```diff
@@ -0,0 +1,252 @@
+``Person`` annotation:
+
+.. code-block:: java
+
+    @PUT
+    public Person replace(@Valid Person person) {
+        // ...
+    }
+
+If the name field is missing, Dropwizard will return a ``422 Unprocessable Entity`` response
+detailing the validation errors: ``name may not be empty``
+
+.. _man-validation-validations-optional-constraints:
+
+``Optional<T>`` Constraints
+***************************
+
+When ``Optional<T>`` is constrained, it is the type contained within the ``Optional`` that is
+constrained. If the ``Optional`` is absent, then the constraints are not applied.
+
+.. note::
+
+    Be careful when using constraints with ``*Param`` annotations with ``Optional<String>`` types as
+    there is a subtle, but important distinction between null and empty. If a client requests
+    ``bar?q=``, ``q`` will evaluate to ``Optional.Of("")``. If you want ``q`` to evaluate to
+    ``Optional.absent()`` in this situation, change the type to ``NonEmptyStringParam``
+
+.. _man-validation-programming-by-contract:
+
+Programming by Contract
+=======================
+
+Validations can be performed on endpoint return values. This enables you, as the service provider,
+to make guarantees. If the constraints are violated, instead of returning faulty data, an internal
+server error is sent. This is known as "Programming by Contract"
+
+Our venerable ``Person`` needs a name and we can guarantee that it will have a name in a response if
+we mark the method with ``@Valid`` or ``@Validated``.
+
+.. code-block:: java
+
+    @PUT
+    @Valid
+    public Person replace(@Valid Person person) {
+        // Oops, we got a bug!
+        return new Person(null);
+    }
+
+In our example, the server will respond a ``500 Internal Server Error`` status and an error message:
+``name may not be empty``.
+
+.. _man-validation-annotations:
+
+Annotations
+===========
+
+In addition to the annotations defined in Hibernate, Dropwizard contains another set of annotations,
+which are briefly shown below.
+
+.. code-block:: java
+
+    public class Person {
+        @NotEmpty
+        private final String name;
+
+        @NotEmpty
+        @OneOf(value = {"m", "f"}, ignoreCase = true, ignoreWhitespace = true)
+        // @OneOf forces a value to value within certain values.
+        private final String gender;
+
+        @Max(10)
+        @Min(0)
+        // The integer contained, if present, can attain a min value of 0 and a max of 10.
+        private final Optional<Integer> animals;
+
+        @JsonCreator
+        public Person(@JsonProperty("name") String name) {
+            this.name = name;
+        }
+
+        @JsonProperty("name")
+        public String getName() {
+            return name;
+        }
+
+        // Method that must return true for the object to be valid
+        @ValidationMethod(message="name may not be Coda")
+        public boolean isNotCoda() {
+            return !(name.equals("Coda"));
```

Comment:
Good catch! I rashly changed the current example in the name of aesthetics, thanks for correction.